src/task_collection.js

```diff
@@ -0,0 +1,204 @@
+    removeBtn.classList.add('remove-btn');
+    removeBtn.id = `remove-btn-${newTask.index}`;
+    removeBtn.innerText = 'Delete';
+    controls.appendChild(removeBtn);
+    const editBtn = document.createElement('button');
+    editBtn.classList.add('edit-btn');
+    editBtn.id = `edit-btn-${newTask.index}`;
+    editBtn.innerText = 'Edit';
+    controls.appendChild(editBtn);
+    const editInput = document.createElement('input');
+    editInput.setAttribute('type', 'text');
+    editInput.className = 'edit-input d-none';
+    editInput.value = newTask.description;
+    editInput.id = `edit-input-${newTask.index}`;
+    listItem.appendChild(editInput);
+    listItem.appendChild(controls);
+    ul.appendChild(listItem);
+
+    // Add event listeners
+    this.addTaskRemoveEventListener(newTask.index);
+    this.addTaskEditBtnEventListener(newTask.index);
+    this.addTaskEditInputEventListener(newTask.index);
+  }
+
+  addTaskRemoveEventListener(index) {
+    const removeBtn = document.querySelector(`#remove-btn-${index}`);
+    removeBtn.addEventListener('click', () => {
+      this.remove(index);
+    });
+  }
+
+  addTaskEditBtnEventListener(index) {
+    const editBtn = document.querySelector(`#edit-btn-${index}`);
+    editBtn.addEventListener('click', () => {
+      const controls = document.querySelector(`#controls-${index}`);
+      controls.classList.add('d-none');
+      const label = document.querySelector(`#label-${index}`);
+      label.classList.add('d-none');
+      const editInput = document.querySelector(`#edit-input-${index}`);
+      editInput.classList.remove('d-none');
+    });
+    this.getCollection();
+  }
+
+  addTaskEditInputEventListener(index) {
+    const editInput = document.querySelector(`#edit-input-${index}`);
+    editInput.addEventListener('keypress', (e) => {
+      if (e.keyCode === 13) {
+        this.editTask(index, e.target.value);
+        e.preventDefault();
+
+        const controls = document.querySelector(`#controls-${index}`);
+        controls.classList.remove('d-none');
+        const label = document.querySelector(`#label-${index}`);
+        label.innerHTML = '';
+        label.classList.remove('d-none');
+        const status = document.createElement('input');
+        status.setAttribute('type', 'checkbox');
+        status.id = `task${index}`;
+        status.setAttribute('name', `status-${index}`);
+        label.appendChild(status);
+        label.append(e.target.value);
+        const editInput = document.querySelector(`#edit-input-${index}`);
+        editInput.classList.add('d-none');
+      }
+    });
+  }
+
+  remove(idx) {
+    this.collection = this.collection.filter(
+      (task) => task.index !== parseInt(idx, 10),
+    );
+
+    populateStorage(this.collection);
+
+    const toRemove = document.querySelector(`#task-${idx}`);
+    toRemove.remove();
+    TaskCollection.counter -= 1;
+    this.resetIndexes(idx);
+    this.render();
+  }
```

Comment:
Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique. That means if there `1, 2, 3, 4, 5` items in the array. If I delete `item 3` then the indexes of the remaining `items` should be reset to `1, 2, 3, 4`. Kindly work on your function to achieve this.

## Your app:
![image](https://user-images.githubusercontent.com/10332499/154387356-4b0d2805-3699-4192-8c16-9e8f7276065a.png)